docker_entrypoint.sh

```diff
@@ -40,189 +62,304 @@ ADV_TOR_MAXSOCKSCONNS=$(yq e '.advanced.tor.maxsocksconns' ${BITMONERO_DIR}/star
- echo    "pad-transactions=1" >> $new_conf_template
+ echo -e "\n# Use Tor's socks proxy for p2p traffic (note: --proxy cannot reach .onion nodes)" >> $config_file
+ echo    "proxy=TOR_SOCKS_PROXY_HOST:TOR_PORT" >> $config_file
+ echo    "# Pad relayed transactions to next 1024 bytes to help defend against traffic volume analysis. This only makes sense if you are behind Tor or I2P." >> $config_file
+ echo    "pad-transactions=1" >> $config_file
 fi
 
 #Gossip config:
 if [ "$ADV_P2P_GOSSIP" = "false" ] ; then
- echo -e "\n# GOSSIP" >> $new_conf_template
- echo "#Tell our peers not to gossip our node" >> $new_conf_template
- echo "hide-my-port=1" >> $new_conf_template
- echo "# Disable UPnP port mapping" >> $new_conf_template
- echo "igd=disabled" >> $new_conf_template
-elif [ "$ADV_P2P_PUBLICRPC" = "true" ] ; then
+ echo -e "\n# GOSSIP" >> $config_file
+ echo "#Tell our peers not to gossip our node w/ P2P port" >> $config_file
+ echo "hide-my-port=1" >> $config_file
+ echo "# Disable UPnP port mapping" >> $config_file
+ echo "igd=disabled" >> $config_file
+elif [ "$ADV_TOR_TORONLY" = "true" ] ; then
+  echo "# Advertise our onion as the reachable host for incoming P2P connections" >> $config_file
+  echo "anonymous-inbound=PEER_TOR_ADDRESS:MONERO_P2P_PORT,MONEROD_LOCAL_HOST:MONERO_P2P_PORT_LOCAL_BIND,ADV_TOR_MAXONIONCONNS" >> $config_file
+  echo "# Disable UPnP port mapping" >> $config_file
+  echo "igd=disabled" >> $config_file
+fi
+
+if [ "$ADV_P2P_PUBLICRPC" = "true" ] ; then
  #RPC config:
- echo -e "\n# PUBLIC RPC" >> $new_conf_template
- echo "# Node advertisement: Requires --restricted-rpc, --rpc-bind-ip and --confirm-external-bind" >> $new_conf_template
- echo "# Advertise to wallets crawling the p2p network that they can use this node as a \"remote node\" for connecting their wallets." >> $new_conf_template
- echo "public-node=1" >> $new_conf_template
- if [ "$ADV_TOR_TORONLY" = "true" ] ; then
-  echo "# Advertise onion as public remote node (Communicated to wallet clients that crawl to our p2p network port, when public-node=1)" >> $new_conf_template
-  echo "anonymous-inbound=RPC_TOR_ADDRESS:MONERO_RPC_PORT_RESTRICTED,MONERO_ANON_INBOUND_HOST:MONERO_RPC_PORT_HS,ADV_TOR_MAXONIONCONNS" >> $new_conf_template
-  echo "# Disable UPnP port mapping" >> $new_conf_template
-  echo "igd=disabled" >> $new_conf_template
- #elif [ "ADV_P2P_UPNP" = "true" ] ; then
- # echo "#Enable UPnP port mapping" >> $new_conf_template
- # echo "igd=enabled" >> $new_conf_template
- fi
+ echo -e "\n# EXTERNAL CONNECTIONS" >> $config_file
+ echo "# Node advertisement: Requires --restricted-rpc, --rpc-bind-ip and --confirm-external-bind" >> $config_file
+ echo "# Advertise to wallets crawling the p2p network that they can use this node as a \"remote node\" for connecting their wallets." >> $config_file
+ echo "public-node=1" >> $config_file
```

Comment:
This req restricted-rpc etc. 
id rm this

Reply:
This is optional (and not the default).  See the `if`s around the part you quoted.  If the user picks it, they intend to have it on and we will turn it on.